File: VBNetCore.Mvvm/Abstractions/IPopupable.vb
Imports VBNetCore.Mvvm.ViewModels

Namespace VBNetCore.Mvvm.Abstractions
    Public Interface IPopupable
        Function ShowPoup(ByVal popupViewModel As PopupViewModel) As PopupViewModel
        Sub HidePopup()
        Property CurrentPopup As PopupViewModel
        Property IsPopupVisible As Boolean
    End Interface
End Namespace

File: VBNetCore.Mvvm/Abstractions/IPopupable.cs
using NetCore.Mvvm.ViewModels;

namespace NetCore.Mvvm.Abstractions
{
    public interface IPopupable
    {
        PopupViewModel? ShowPoup(PopupViewModel popupViewModel);
        void HidePopup();
        PopupViewModel? CurrentPopup { get; set; }
        bool IsPopupVisible { get; set; }
    }
}
